recipe/meta.yaml

```diff
@@ -36,128 +43,142 @@ source:
 
 build:
   number: {{ build }}
-  string: cuda{{ cuda_compiler_version | replace('.', '') }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
-  string: cpu_{{ blas_impl }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                 # [cuda_compiler_version == "None"]
-  detect_binary_files_with_prefix: false
-  run_exports:
-    - {{ pin_subpackage('libtorch', max_pin='x.x') }}
-  ignore_run_exports_from:
-    - python *                               # [megabuild]
-    - numpy *                                # [megabuild]
-    - cross-python_{{ target_platform }}     # [megabuild and build_platform != target_platform]
-  ignore_run_exports:
-    - python *                               # [megabuild]
-    - numpy *                                # [megabuild]
```

Comment:
Why all these changes?

Reply:
It's just an indentation change. If you switch off whitespace visibility, you'll see what I mean.

Reply:
See in particular: https://github.com/conda-forge/pytorch-cpu-feedstock/pull/238/commits/981a04d378bda35a2f2852cd0a448f0bcc9d56e6?w=1

Reply:
No, it's not only an indentation change. You are changing the top level recipe from `libtorch` to `pytorch-recipe` and then moving build scripts to an output. We need a fan-out build here and it's only a coincidence that the recipe in this PR works. Please revert.

Reply:
Fan-outs also work between outputs, a lot of other recipes rely on that (e.g. arrow).

I'm really against reusing the same name for both package & output. It's really non-obvious what's happening, and it's much better to create a separate output.

Reply:
I'm against relying on an undocumented feature where there's an documented feature to be used.

Reply:
Can you point me to some documentation for which fan-out builds are supported (or not)?

It's been working like that for years and has several users that I'm aware of. I'd certainly raise an issue in conda-build as a regression if any of this breaks. It's a natural application of the variant algebra from my POV. One key (`python:`) with several entries is used for one output but not the other, so build that one first and then loop through the python versions for the dependent one.

In contrast, the way that reusing the package name for an output actually works is all but obvious for anyone who's not already familiar with this (very rare) pattern - e.g. where are the tests (not defined before this PR, but if they were, they'd come before the output they're testing). A lot of people will look at the pytorch feedstock. It's more important to me that the recipe be understandable than how conda-build does it behind the scenes.

Reply:
Can you please stop arguing over every little thing? You are making conda-forge less fun for me and to at least one other person I know of. I'm going to stop commenting here. I'll leave it upto @conda-forge/pytorch-cpu to merge this. Please add yourself as a maintainer here as I will not contribute to this feedstock anymore.

Reply:
> Can you please stop arguing over every little thing?

Excuse me? You opened this discussion and insisted on your view. Am I not allowed to defend mine?

> You are making conda-forge less fun for me and to at least one other person I know of.

I'm sorry to hear that. As you well know, I've reached out to you repeatedly over the last couple of years to find a way to work together productively, asking what you'd like me (not) to do, and you've given me no indication besides (paraphrasing) "do your homework better". While I've tried that to the absolute limit my free time allows, it's frankly been challenging to work with your curt-verging-on-combative responses towards me across many a feedstock...

> Please add yourself as a maintainer here as I will not contribute to this feedstock anymore.

... but yet I keep trying, because you are an absolutely essential contributor/reviewer. So I'll continue trying to do my best to find ways to work together harmoniously. Still, I don't see why we cannot resolve a difference of opinion through a technical discussion (rather than throwing out the baby with the bathwater), or why your opinion (on a feedstock where we're both not officially maintainers) should win by default over mine.

Reply:
@isuruf I ask that you please reconsider your statement:

> as I will not contribute to this feedstock anymore.

it has been great to work with you on this project (and many others).

I think that review comments are often difficult to contextualize from people sitting on the other side of a screen, often on the other side of world. I know that even short sentences like:

> Why all these changes?

and 

> It's just an indentation change [...]

were well thought out by the both of you. However, after a long day of work, I know that both of these might seem like you are brushing off the comment of the other.

I may be "injecting" my own thoughts in here, but my reading of your interactions make it seem like the fuller sentences would have been:

> Interesting approach, I'm surprised that this isn't causing a full rebuild of `libtorch` many times. It seems that the fan-out build is still being triggered even though it isn't at the top level. I don't think this is well documented and may disappear in a future version of conda-build. Is it possible to the old syntax? If so I would prefer that approach.

and

> I've been using fan-outs in many other recipes such as grpc, if this is a quirk in the conda-build, I would rather get it resolved. May you point me to the documentation you are referring to so that I may appropriately ask for clarification in a centralized location?


While it may be hard to reconcile, I know you are both going above an beyond with your personal time to review these recipes and keep conda-forge as a whole in good working order.


----------

Personally, I've avoided these so called "fan-out" builds and historically have been moving away from naming multi-output recipes the same name as one of the outputs due to the rapid pace of their development and the fact that they seem to change feature set every so often (in my perspective)


 I tried to search through both:
* https://docs.conda.io/projects/conda-build/en/stable/
* https://conda-forge.org/docs/

for the word "fan-out" and I couldn't find anything of substance. It would be helpful for me. I agree with Isuru's thoughts that we should use "documented" features (especially for these nearly impossible to build recipes), and I feel like a comment pointing to the appropriate literature might be appropriate at the top level of the recipe.


As a way to move forward could:

* h-vetinari is it possible to move to the old syntax for now on this recipe?
* isuruf are you able to point us to the documentation on fan-out builds so that we can refer to it here and ask questions in the most appropriate repo?

Reply:
Thanks for the mediation effort @hmaarrfk! I can go back to the previous way for now, and refer to documentation, if any exists (I'm not aware of one, so I consider both ways equally undocumented, but if that's what's necessary to move on, so be it).

Otherwise we can open an issue in conda-build for clarification.